### PR TITLE
Spectator camera with HUD

### PR DESCRIPTION
**Pull request by wima12**
Added an entity SpectatorCamera.xml contains a SpectatorCamera. If it is present in a level, then any player that connects and picks a team will see everything from the SpectatorCamera. When a player dies they will see their explosioneffect and then swap to the SpectatorCamera.

NewMapWSpectatorCam.xml can be used to test the functionality.

### CODE REVIEW

File: resources/DefaultConfig.ini [hunk 1]
```diff
@@ -4,7 +4,7 @@ LoadMap=
 ; if true -> Pool allocation is not used when calling Allocate/Free, just use regular dynamic allocation. 
 ; if false -> Use pool allocation.
 DisableMemoryPool=false
-RespawnTime = 8.0
```
Comment (JockeBTH) [inline resources/DefaultConfig.ini:7]: is -1.0 suppose to be the default value?

File: src/Engine/Core/Util/Logging.cpp [hunk 2]
```diff
@@ -33,8 +33,8 @@ void _LOG(_LOG_LEVEL logLevel, const char* file, const char* func, unsigned int
 	va_end(args);
 
 	if (logLevel == LOG_LEVEL_ERROR) {
-		//std::cerr << file << ":" << line << " " << func << std::endl;
-		//std::cerr << _LOG_LEVEL_PREFIX[logLevel] << message << std::endl;
+		std::cerr << file << ":" << line << " " << func << std::endl;
```
Comment (JockeBTH) [inline src/Engine/Core/Util/Logging.cpp:36]: Was this for debug or intended? :)

File: resources/DefaultConfig.ini (hunk 1 above)
Comment (wima12) [inline resources/DefaultConfig.ini:7]: Yes, the default behavior should be not to override the component in the map with the config value. Any negative value is the same as having no value at all, i.e. take the CapturePointGameMode value.

File: src/Engine/Core/Util/Logging.cpp (hunk 2 above)
Comment (wima12) [inline src/Engine/Core/Util/Logging.cpp:36]: I just rectified something someone else did there. Apparently someone commented out all logging for the ERROR level. I do think we want to see error messages in the console.

**Comment by JockeBTH** [comment]
Looks good man! :us: :hamburger:  /play vuvuzela :sunglasses:

**Pull request merged**